packages/tools/src/utilities/math/polyline/isPointInsidePolyline.ts

```diff
@@ -0,0 +1,98 @@
+import type { Types } from '@cornerstonejs/core';
+
+/**
+ * Determines whether a 2D point is inside a polyline.
+ * The algorithm works by drawing horizontal rays from the point to the right side
+ * of the polygon, counting the number of times the ray intersects a polygon edge.
+ *  If the winding number is odd, the point is inside the polygon.
+ *
+ * @param point - The 2D point to check.
+ * @param polyline - The polyline represented as an array of 2D points.
+ * @returns A boolean indicating whether the point is inside the polyline.
+ */
+export function isPointInsidePolyline2D(
+  point: Types.Point2,
+  polyline: Types.Point2[]
+) {
+  const n = polyline.length;
+  let windingNumber = 0;
+  const x = point[0],
+    y = point[1];
+
+  for (let i = 0; i < n; i++) {
+    const j = (i + 1) % n;
+    const xi = polyline[i][0],
+      yi = polyline[i][1];
+    const xj = polyline[j][0],
+      yj = polyline[j][1];
+
+    if (yi === yj && y !== yi) {
+      continue;
+    }
+
+    if (
+      ((yi < y && yj >= y) || (yj < y && yi >= y)) &&
```

Comment:
This is a little opaque here - I can figure out what you are doing, but would like a reference to the algorithm in the docs above, or an explanation here - how about:
// If y is between yi and yj, and x is to the right of the line between (xi,yi) and (xj,yj), 
// then the line between (xi,yi) and (xj,yj) is a relevant boundary for being inside/outside the point.

Reply:
I switched the what Leo implemented so all of this can be resolved